ocaml/src/asd_server.ml

```diff
@@ -360,8 +368,41 @@ let execute_query : type req res.
         )
       >>= fun (d,res) ->
       AsdStatistics.new_multi_get stats d;
-      Lwt.return res
-      end
+      return res
+    | MultiGet2 -> fun keys ->
+      let laters = ref [] in
+      AsdStatistics.with_timing_lwt
+        (fun () ->
+         Lwt_log.ign_debug_f "MultiGet2 for %s" ([%show: Slice.t list] keys);
+         (* first determine atomically which are the relevant Value.t's *)
+         List.map
+           (fun k ->
+            get_value_option kv k
+            |> Option.map
+                 (fun (cs, blob) ->
+                  let b = match blob with
+                    | Value.Direct s -> Asd_protocol.Value.Direct s
+                    | Value.OnFs (fnr, size) ->
+                       laters := (fnr, size) :: !laters;
+                       Asd_protocol.Value.Later size in
+                  b, cs))
+           keys
+         |> Lwt.return)
+      >>= fun (d,res) ->
+      (* AsdStatistics.new_multi_get stats d; *)
```

Comment:
This duration doesn't contain everything yet ... that's why I commented it while working on this branch (and then forgot it afterwards).
But actually the same can be said for the duration for multiget on https://github.com/openvstorage/alba/pull/1/files#diff-8af52f72b48bb67ec73232134ae73a9fR369 , because that one doesn't take into account putting the value(s) on the wire.

Actually for all these commands we measure the duration twice, once here, and another time at 
https://github.com/openvstorage/alba/pull/1/files#diff-8af52f72b48bb67ec73232134ae73a9fR876 .
We probably only want to keep this last one (it includes the entire duration), and have it update the statistics.

I'll update that tomorrow.